spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/core/QueueMessageUtils.java

```diff
@@ -38,6 +48,23 @@
     private static final String RECEIPT_HANDLE_MESSAGE_ATTRIBUTE_NAME = "ReceiptHandle";
     private static final String MESSAGE_ID_MESSAGE_ATTRIBUTE_NAME = "MessageId";
 
+    private static final Map<Class<? extends Number>, Class<? extends Number>> PRIMITIVE_NUMBER_TYPE_TO_WRAPPED_TYPE_MAP = Stream.of(
```

Comment:
I didn't know about this before, but checkout `org.springframework.util.NumberUtils`. Looks like it will handle this without much custom code.

Reply:
`org.springframework.util.NumberUtils#parseNumber(String, Class)` unfortunately doesn't support primitive types (e.g., `int.class`), only the wrapped types—it'll throw an exception.

That's why I added a map to look up the wrapped type given a primitive type (and unfortunately there doesn't seem to be a simple to do it without relying on Apache Commons or Google Guava or other libraries that provided a primitive -> wrapped type util method).

We could eliminate a majority of this look up stuff if we just don't support stuff like `"Number.int"`, etc., and only permit the wrapped types, e.g., `"Number.Integer"`. What do you think?

Reply:
@spencergibb Any update on this? What would you prefer?

Reply:
Note there's another open PR (#293 ) to add support for primitive types.